packages/engine/src/output/buffer/util.rs

```diff
@@ -1,42 +1,83 @@
 };
 
+// TODO: move this to top-level
+#[derive(PartialEq, Eq, Clone)]
+pub enum EngineExitStatus {
+    Success,
+    Error,
+}
+
 // TODO: move this to top-level
 /// Shared memory cleanup in the process hard crash case.
 /// Not required for pod instances.
-pub fn cleanup_experiment(experiment_id: &ExperimentId) -> Result<()> {
+pub fn cleanup_experiment(
+    experiment_id: &ExperimentId,
+    exit_status: EngineExitStatus,
+) -> Result<()> {
     MemoryId::clean_up(experiment_id)?;
 
-    // TODO: We don't want to be deleting the parts files by default. We should figure out what to
-    //   do with this.
-    // remove_experiment_parts(experiment_id).or_else(|e| {
-    //     let err_string = e.to_string();
-    //     if err_string.contains("kind: NotFound") {
-    //         Ok(())
-    //     } else {
-    //         Err(Error::from(format!(
-    //             "Error while trying to remove experiment parts folder: {err_string}"
-    //         )))
-    //     }
-    // })?;
+    // Cleanup python socket files in case the engine didn't
+    let frompy_files = glob::glob(&format!("{experiment_id}-frompy*"))
+        .map_err(|err| Error::Unique(format!("cleanup glob error: {err}")))?;
+    let topy_files = glob::glob(&format!("{experiment_id}-topy*"))
+        .map_err(|err| Error::Unique(format!("cleanup glob error: {err}")))?;
+
+    frompy_files
+        .chain(topy_files)
+        .filter_map(std::result::Result::ok)
+        .for_each(|path| match std::fs::remove_file(&path) {
+            Ok(_) => {
+                match exit_status {
+                    EngineExitStatus::Success => tracing::event!(
+                        tracing::Level::ERROR,
+                        "Removed file {path:?} that should've been cleanup by the engine."
+                    ),
+                    EngineExitStatus::Error => tracing::event!(
+                        tracing::Level::TRACE,
+                        "Removed file {path:?} that should've been cleanup by the engine."
+                    ),
+                };
+            }
+            Err(err) => {
+                tracing::warn!("Could not clean up {path:?}: {err}");
+            }
+        });
+
+    remove_experiment_parts(experiment_id)?;
 
     Ok(())
 }
 
-#[allow(dead_code)]
 fn remove_experiment_parts(experiment_id: &ExperimentId) -> Result<()> {
-    let mut base_path = PathBuf::from(RELATIVE_PARTS_FOLDER);
-    // TODO: this is unused at the moment so it's fine, but this logic is wrong, we name our folders
-    //  differently, we should update the design to store the paths and use them here when we use
-    //  the clean up code again
-    base_path.push(experiment_id.to_string());
-    tracing::trace!("Removing all parts files in: {base_path:?}");
-    std::fs::remove_dir_all(base_path)?;
+    let parts_fimes = glob::glob(&format!("{RELATIVE_PARTS_FOLDER}/{experiment_id}"))
+        .map_err(|err| Error::Unique(format!("cleanup glob error: {err}")))?;
```

Comment:
Why do we need a `glob` here? Can't we simply remove the `parts/{experiment_id}` folder?

Reply:
Removed in https://github.com/hashintel/hash/pull/477/commits/7e2bcf26c3eef091683e8e5ba14c8e622acb5e33